packages/core/src/conflict-resolution.ts

```diff
@@ -253,18 +240,14 @@ export class ConflictResolution {
    */
   private async applyLogEntryToState<T extends Stream>(
     entry: CommitData, state: StreamState, handler: StreamHandler<T>): Promise<StreamState> {
-    const logEntry = await this.getCommitData(entry)
-    const commitMeta = { cid: logEntry.cid, timestamp: logEntry.timestamp }
-    if (StreamUtils.isAnchorCommit(logEntry.commit)) {
+    const logEntry = await Utils.getCommitData(entry, this.dispatcher)
```

Comment:
Hmm, 

> overall this is looking great. I'm very happy to have killed off CommitMeta and merged it with CommitData.
> 
> My number 1 remaining concern is that we can't rely on the type system right now to know whether the underlying commit data has been loaded or not. Whenever you get a CommitData argument, it might or might not have the underlying commit loaded into it. Some places in the code (such as the streamtype handlers) now assume that what it's given has the underlying commit data in it, while other places that receive CommitData arguments does not, and calls `getCommitData` to get it even though the argument was already a CommitData.
> 
> I think if we make the relevant CommitData fields (`commit` and `type` at a minimum) required, then we can trust that any place we are given CommitData we actually have that data present, and then if there are places that sometimes have that data present and sometimes not, their type signature could use `LogEntry | CommitData` to note that.

Good thoughts, thanks for thinking about this @stbrody.

The problem with making `type` required is that in places like `ConflictResolution.fetchLog()` we don't know the `type` because we don't have the `commit` yet.

This is going to be a general problem - we need the `commit` to determine the `type` and the only way to make both required would be to fetch the `commit` before the call to `getCommitData()`, which would require leaving an IPFS lookup outside of `Utils.getCommitData()`.

`in-memory-anchor-service.ts` had the same problem of having just the `cid` but not `type` or `commit` but I was able to revert the file to use the old approach to fetch the `commit` separately and determine the `type`. Since it is a test file, this should be ok, although I don't like leaving any IPFS lookups lingering outside of `Utils.getCommitData()` if it can be helped.

Similarly, in `Repository.fromNetwork()`, we have the `cid` and the `type` (genesis) but not the `commit`. Retrieving the `commit` before constructing the `CommitData` object would defeat the purpose of collapsing all lookups inside `Utils.getCommitData()`.

Alternatively, we could made `Utils.getCommitData()` accept `LogEntry | CommitData` but then it would be confusing to reliably determine what information was already available and what needed to be looked up.

`commit` being optional in `commitData` allows us to reliably determine the latter is incomplete and that the `commit`, `envelope`, `proof`, `timestamp` need to be looked up.